consul/acl.go

```diff
@@ -366,6 +370,42 @@ func (f *aclFilter) filterNodeDump(dump *structs.NodeDump) {
 	*dump = nd
 }
 
+// filterPreparedQueries is used to filter prepared queries based on ACL rules.
+// We prune entries the user doesn't have access to, and we redact any tokens
+// unless the client has a management token. This eases the transition to
+// delegated authority over prepared queries, since it was easy to capture
+// management tokens in Consul 0.6.3 and earlier, and we don't want to
+// willy-nilly show those. This does have the limitation of preventing delegated
+// non-management users from seeing captured tokens, but they can at least see
+// that they are set and we want to discourage using them going forward.
+func (f *aclFilter) filterPreparedQueries(queries *structs.PreparedQueries) {
+	isManagementToken := f.acl.ACLList()
+	ret := make(structs.PreparedQueries, 0, len(*queries))
+	for _, query := range *queries {
+		if !f.acl.PreparedQueryRead(query.GetACLPrefix()) {
+			f.logger.Printf("[DEBUG] consul: dropping prepared query %q from result due to ACLs", query.ID)
+			continue
+		}
+
+		// Secure tokens unless there's a management token doing the
+		// query.
+		if isManagementToken || query.Token == "" {
```

Comment:
This seems like a sort of odd exception - Do you think it would make sense to just always mask it if you aren't using a management token?